packages/components/src/Grid.astro

```diff
@@ -0,0 +1,219 @@
+---
+const { container, fluid, xs, sm, md, lg, auto, gap } = Astro.props;
+const containerClass = container ? "grid-container" : "";
+const autoClass = auto ? "auto-columns" : "";
+const gapClass = gap ? `gap-${gap}` : "";
+const xsClass = xs ? `col-${xs}` : "";
+const smClass = sm ? `col-sm-${sm}` : "";
+const mdClass = md ? `col-md-${md}` : "";
+const lgClass = lg ? `col-lg-${lg}` : "";
+const fluidClass = fluid ? "fluid" : "";
```

Comment:
```suggestion
const containerClass = container && "grid-container";
const autoClass = auto && "auto-columns";
const gapClass = gap && `gap-${gap}`;
const xsClass = xs && `col-${xs}`;
const smClass = sm && `col-sm-${sm}`;
const mdClass = md && `col-md-${md}`;
const lgClass = lg && `col-lg-${lg}`;
const fluidClass = fluid && "fluid";
```

Because `classMerge` filters out falsy values we can simplify this a bit.